### ISSUE TITLE
Cast-web syncing

### ISSUE THREAD

**Issue by gietlt** [vervallsweg/cast-web-api#108]
I have preset 1 as classic rock, preset 2 as country.  I can hit any of the presets and the stop and it works fine.  When it comes to play and next or previous track, I can only execute one command at a time then I have to hit the gear key in the the app then save the config.  Example: I can hit the play button, then i cant  use the next or previous track and the pause button does not show.  Then i hit the gear key and save.  Now I have the pause key.  I now can hit the pause, next track or previous track, after that I have to hit the gear key and save before I can use it again.  I found this issue because I cant control the speaker through action tiles.

**Comment by cgmckeever** [comment]
@gietlt 👋 .. OK .. Im tracking some more here. Let me make sure .... Once you start a preset, you can do one following action and then cast-web sort of stalls out (but continues to broadcast). To get back working, you need to open settings and save - but then you are only able to do one more action - and then you need to repeat the setting save -- and its a big circle?

I wrestled with presets, finally got one working, but I havent actually tried anything other than stop ;) .. BUT I do sort of feel that 9/10 after I hit stop, cast-web gets a little weird and I have to restart things. I need to test it a little more ... Can you post your preset snippet here? I can try it as well.

In addition, without any presets, if you cast something to the speaker, do all the buttons work repeatably without only getting one action and then you need to reset things?

**Comment by gietlt** [comment]
Yes.  That is exactly what happens.  Ice bear asked the same question without presets.  I thought you need presets to make it work.

Here is my snippet

{"preset1":{"mediaTitle":"Classic rock","mediaSubtitle":"Rock","mediaType":"audio/mp3","mediaUrl":"http://us4.internet-radio.com:8258/;stream","mediaStreamType":"LIVE","mediaImageUrl":""},"preset2":{"mediaTitle":"Country","mediaSubtitle":"Country","mediaType":"audio/mp3","mediaUrl":"http://us3.internet-radio.com:8038/;stream","mediaStreamType":"","mediaImageUrl":""},"preset3":{"mediaTitle":"X-mas","mediaSubtitle":"X-mas","mediaType":"audio/mp3","mediaUrl":"http://bitsmitter.com:9008/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset4":{"mediaTitle":"50s and 60s rock","mediaSubtitle":"50s and 60s rock","mediaType":"audio/mp3","mediaUrl":"http://167.114.64.181:8360/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset5":{"mediaTitle":"Preset 5","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""},"preset6":{"mediaTitle":"Preset 6","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""}}

**Comment by cgmckeever** [comment]
No - you dont need presets for it to work. But it is a nice feature. I just confirmed, that once I start playing. I only get one action then I have to futz with it to make the commands work and such. I really dont use the UI to do anything so Im not going to be much more of a help other than at least helping you confirm to some degree that it isnt your set up .. !

**Comment by cgmckeever** [comment]
Yeah, I dont know whats going on with the UI controls. They are very peculiar

**Comment by gietlt** [comment]
I don't know if this helps, but I tried using the snippet from KTDT00 in the cast-web1.1.0 community stream on entry 905 as a template.  I copied the snippet into my peaker preset object in the cast-web manager then hit next, next and save expecting to see preset1 as dingdong.  The snippet did not save.  It made everything blank as if I just loasded the device.  Here is the snippet from KTDT00

{“preset1”:{“mediaTitle”:“DingDong”,“mediaSubtitle”:“Doorbell Sound”,“mediaType”:“audio/mp3”,“mediaUrl”:“http://www.sounds.beachware.com/2illionzayp3may/joptlz/DINGDONG.mp3",“mediaStreamType”:“LIVE”,“mediaImageUrl”:""},“preset2”:{“mediaTitle”:"Preset 2”,“mediaSubtitle”:"",“mediaType”:"",“mediaUrl”:"",“mediaStreamType”:"",“mediaImageUrl”:""},“preset3”:{“mediaTitle”:“Preset 3”,“mediaSubtitle”:"",“mediaType”:"",“mediaUrl”:"",“mediaStreamType”:"",“mediaImageUrl”:""},“preset4”:{“mediaTitle”:“Preset 4”,“mediaSubtitle”:"",“mediaType”:"",“mediaUrl”:"",“mediaStreamType”:"",“mediaImageUrl”:""},“preset5”:{“mediaTitle”:“Preset 5”,“mediaSubtitle”:"",“mediaType”:"",“mediaUrl”:"",“mediaStreamType”:"",“mediaImageUrl”:""},“preset6”:{“mediaTitle”:“Preset 6”,“mediaSubtitle”:"",“mediaType”:"",“mediaUrl”:"",“mediaStreamType”:"",“mediaImageUrl”:""}}

**Comment by cgmckeever** [comment]
Yeah, there is probably a format issue. I wrestled with presets never appearing which also turned out to be improper JOSN format .. I think your issue is more with cast web than anything

**Comment by jager181** [comment]
> Yes. That is exactly what happens. Ice bear asked the same question without presets. I thought you need presets to make it work.
> 
> Here is my snippet
> 
> {"preset1":{"mediaTitle":"Classic rock","mediaSubtitle":"Rock","mediaType":"audio/mp3","mediaUrl":"[http://us4.internet-radio.com:8258/;stream","mediaStreamType":"LIVE","mediaImageUrl":""},"preset2":{"mediaTitle":"Country","mediaSubtitle":"Country","mediaType":"audio/mp3","mediaUrl":"http://us3.internet-radio.com:8038/;stream","mediaStreamType":"","mediaImageUrl":""},"preset3":{"mediaTitle":"X-mas","mediaSubtitle":"X-mas","mediaType":"audio/mp3","mediaUrl":"http://bitsmitter.com:9008/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset4":{"mediaTitle":"50s](http://us4.internet-radio.com:8258/;stream%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset2%22:%7B%22mediaTitle%22:%22Country%22,%22mediaSubtitle%22:%22Country%22,%22mediaType%22:%22audio/mp3%22,%22mediaUrl%22:%22http://us3.internet-radio.com:8038/;stream%22,%22mediaStreamType%22:%22%22,%22mediaImageUrl%22:%22%22%7D,%22preset3%22:%7B%22mediaTitle%22:%22X-mas%22,%22mediaSubtitle%22:%22X-mas%22,%22mediaType%22:%22audio/mp3%22,%22mediaUrl%22:%22http://bitsmitter.com:9008/;stream/1%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset4%22:%7B%22mediaTitle%22:%2250s) and 60s rock","mediaSubtitle":"50s and 60s rock","mediaType":"audio/mp3","mediaUrl":"[http://167.114.64.181:8360/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset5":{"mediaTitle":"Preset](http://167.114.64.181:8360/;stream/1%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset5%22:%7B%22mediaTitle%22:%22Preset) 5","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""},"preset6":{"mediaTitle":"Preset 6","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""}}

Hey, where did you find those usable media links?

**Comment by jager181** [comment]
Is there documentation that goes over how to find set up a preset? I have not been able to set up a station.  Even with public streaming stations like: https://www.internet-radio.com/

**Comment by cgmckeever** [comment]
@jager181 it takes some googling, there isnt a really good one-stop-shop answer for it. For my local NPR I googled for a while, and then wond up finding something on REDDIT

**Comment by jager181** [comment]
Any advice on what to start googling?

**Comment by cgmckeever** [comment]
`name of station` stream url was where I started, I then just followed the rabbit hole of the keywords that started to appear (mainly around what those links are called) ..

**Comment by gietlt** [comment]
Is there a way to uninstall web-cast-api on raspberry pi?  I would like to see if uninstall and reinstall it may fix my issue

**Comment by jager181** [comment]
You can try and do 
apt list --installed 
and remove anything associated with that.

**Comment by cgmckeever** [comment]
`npm uninstall cast-web-api` ?

**Comment by jager181** [comment]
I don't believe it installs locally, you would just remove the packages npm, node, etc. and you can delete the folders installed.

**Comment by gietlt** [comment]
This is driving me crazy.  I removed node.js, npm and cast-web-api.  Deleted app.  Reinstalled everything.  Same out come.  I am running node v8.  Is this correct or could this be my problem

**Comment by cgmckeever** [comment]
@gietlt Im pretty sure what you are seeing is an issue with cast web itself. Im experiencing the same thing ..

**Comment by gietlt** [comment]
cast-web-api on my raspberry pi or cast-web service manager app, or either one?

**Comment by cgmckeever** [comment]
The device handler in the ST app itself. Im pretty sure that the cast-web-api is solid. Ive used it a .ot with webcore, and it does what it needs to all the time. Using the device in the ST app just seems clunky, especially with the exact issue you are talking about - freezing with presets

**Comment by gietlt** [comment]
> > Yes. That is exactly what happens. Ice bear asked the same question without presets. I thought you need presets to make it work.
> > Here is my snippet
> > {"preset1":{"mediaTitle":"Classic rock","mediaSubtitle":"Rock","mediaType":"audio/mp3","mediaUrl":"[http://us4.internet-radio.com:8258/;stream","mediaStreamType":"LIVE","mediaImageUrl":""},"preset2":{"mediaTitle":"Country","mediaSubtitle":"Country","mediaType":"audio/mp3","mediaUrl":"http://us3.internet-radio.com:8038/;stream","mediaStreamType":"","mediaImageUrl":""},"preset3":{"mediaTitle":"X-mas","mediaSubtitle":"X-mas","mediaType":"audio/mp3","mediaUrl":"http://bitsmitter.com:9008/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset4":{"mediaTitle":"50s](http://us4.internet-radio.com:8258/;stream%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset2%22:%7B%22mediaTitle%22:%22Country%22,%22mediaSubtitle%22:%22Country%22,%22mediaType%22:%22audio/mp3%22,%22mediaUrl%22:%22http://us3.internet-radio.com:8038/;stream%22,%22mediaStreamType%22:%22%22,%22mediaImageUrl%22:%22%22%7D,%22preset3%22:%7B%22mediaTitle%22:%22X-mas%22,%22mediaSubtitle%22:%22X-mas%22,%22mediaType%22:%22audio/mp3%22,%22mediaUrl%22:%22http://bitsmitter.com:9008/;stream/1%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset4%22:%7B%22mediaTitle%22:%2250s) and 60s rock","mediaSubtitle":"50s and 60s rock","mediaType":"audio/mp3","mediaUrl":"[http://167.114.64.181:8360/;stream/1","mediaStreamType":"LIVE","mediaImageUrl":""},"preset5":{"mediaTitle":"Preset](http://167.114.64.181:8360/;stream/1%22,%22mediaStreamType%22:%22LIVE%22,%22mediaImageUrl%22:%22%22%7D,%22preset5%22:%7B%22mediaTitle%22:%22Preset) 5","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""},"preset6":{"mediaTitle":"Preset 6","mediaSubtitle":"","mediaType":"","mediaUrl":"","mediaStreamType":"","mediaImageUrl":""}}
> 
> Hey, where did you find those usable media links?

I found the tippet from https://community.smartthings.com/t/release-cast-web-1-1-0-chromecast-device-handler-smartapps/88038/928

From ktdt00
TODD on entry line 904

**Comment by gietlt** [comment]
> @gietlt  .. OK .. Im tracking some more here. Let me make sure .... Once you start a preset, you can do one following action and then cast-web sort of stalls out (but continues to broadcast). To get back working, you need to open settings and save - but then you are only able to do one more action - and then you need to repeat the setting save -- and its a big circle?
> 
> I wrestled with presets, finally got one working, but I havent actually tried anything other than stop ;) .. BUT I do sort of feel that 9/10 after I hit stop, cast-web gets a little weird and I have to restart things. I need to test it a little more ... Can you post your preset snippet here? I can try it as well.
> 
> In addition, without any presets, if you cast something to the speaker, do all the buttons work repeatably without only getting one action and then you need to reset things?

I'm not a programmer so I maybe way off here but is there a way to put a save command in the program any time you hit play, pause, next track, previous track or any of the preset buttons.

**Comment by gietlt** [comment]
I'm done with this.  Today I purchased a new SD card and started from scratch.  Same issue.  I made a donation because it worked flawlessly with action tiles to all my Google speakers.  My wife loved how easy it was to control.  Then I upgraded the cast-web-api and it worked for a week.  Now I can't get it to sync at all.  I will put it to a waste of money

**Comment by vervallsweg** [comment]
> I'm done with this. Today I purchased a new SD card and started from scratch. Same issue. I made a donation because it worked flawlessly with action tiles to all my Google speakers. My wife loved how easy it was to control. Then I upgraded the cast-web-api and it worked for a week. Now I can't get it to sync at all. I will put it to a waste of money

Are you on the st community forum? I'm just catching up to things. DM me there, or some other sort of private communication, if this is still an issue. Yes documentation is still planned, but so little time...

**Comment by gietlt** [comment]
Okay.  It works.  I bought the Rachio sprinkler controller.  When I went to add it on the app as a device, the cast-web-api was sitting in there as a device that was not saved.  I saved it and everything works now.  I don't remember having to do that at first installation, but it works.  I'm happy.  I just upgraded to v1.2.0

**Comment by gietlt** [comment]
> > I'm done with this. Today I purchased a new SD card and started from scratch. Same issue. I made a donation because it worked flawlessly with action tiles to all my Google speakers. My wife loved how easy it was to control. Then I upgraded the cast-web-api and it worked for a week. Now I can't get it to sync at all. I will put it to a waste of money
> 
> Are you on the st community forum? I'm just catching up to things. DM me there, or some other sort of private communication, if this is still an issue. Yes documentation is still planned, but so little time...

Tried PM and replying to your email.  I don't think its working.  I have everything working with presets.  I cant get the TTS to work.  I have a calendar notice every other week for recycle.  When the switch turns on I don't get the voice reminder.  Any help you can provide would be greatly appreciated.[
![1561237243043624289774362678600 (002)](https://user-images.githubusercontent.com/16898173/60436045-657d8480-9bc8-11e9-90ab-dd884bfdb7b9.jpg)
![Screenshot_20190603-112017_SmartThings](https://user-images.githubusercontent.com/16898173/60436046-657d8480-9bc8-11e9-990d-93dad2d30dfe.jpg)
![Screenshot_20190603-113111_SmartThings](https://user-images.githubusercontent.com/16898173/60436047-657d8480-9bc8-11e9-9bba-57af7be12d76.jpg)
